ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiCommandService.scala

```diff
@@ -66,7 +67,8 @@ private[apiserver] final class ApiCommandService private (
 
   private val logger = ContextualizedLogger.get(this.getClass)
 
-  private val submissionTracker: TrackerMap = TrackerMap(configuration.retentionPeriod)
+  private val submissionTracker: TrackerMap[TrackerKey] =
+    new TrackerMap(configuration.retentionPeriod, getTrackerKey, newTracker)
```

Comment:
any reason to not receive this through the constructor? we would be able to extract `newTracker` as well then

Reply:
The `submissionTracker`? Works for me.

Reply:
I pulled it out, but moved the scheduled task inwards, into `TrackerMap.selfCleaning`.